### ISSUE TITLE
Question: John is designing a new function

### ISSUE THREAD

**Issue by svpino** [underfitted/bnomial#32]
### John is designing a new function

John knows that the entire goal of gradient descent is to minimize the value of a function. The lower its value, the better the model will be. John read about mean squared error (MSE) and how to use it for this purpose. Mean absolute error (MAE) seems to be another good alternative.

But John has been thinking about designing a different function for his use case:

He wants to use the inverse of the model's accuracy as the loss function. This way the model will try to minimize the number of mistakes it makes by looking directly at the accuracy of the predictions.

Unfortunately, he soon discovered that this approach doesn't work. 

**What was the problem?**

1. Accuracy is not a differentiable function, so it can't be optimized using gradient descent.
2. It wouldn't make sense to optimize the inverse of the accuracy. Instead, John should try to turn the algorithm into a maximization problem and work directly with the original value.
3. Minimizing the inverse of the accuracy is a very slow process because of the extra computations needed to compute the final value.
4. Gradient descent only works with a predefined set of loss functions.

```
Answer: 1
```
